packages/components/src/mobile/inserter-button/index.native.js

```diff
@@ -80,6 +87,9 @@ class MenuItem extends Component {
 							isClipboardBlock && clipboardBlockStyles,
 						] }
 					>
+						{ item.isNew && (
```

Comment:
👋 @dcalhoun, I was just curious whether there might be an opportunity to use [the badge component that was recently created](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/mobile/badge/README.md) for featured images here? I checked and see that wrapping the `<Badge>` component around `<View style={ modalIconStyle }>` (as well as passing a smaller `margin` and `font-size`) yields similar results for the badge: 

![Screenshot_1625153118](https://user-images.githubusercontent.com/2998162/124150082-21f4d400-da89-11eb-9f68-01c7c5fe98c0.png)

This is definitely a question (rather than an informed opinion) on my part, I'm curious to hear whether there'd be disadvantages to using that component vs. the current approach. :)

Reply:
Thank you for the suggestion, @SiobhyB! I agree that reusing a shared badge could make a lot of sense for consistency and avoid duplicative code. 

I had actually _briefly_ tried your `Badge` component while I was building, but ran into the positioning and sizing discrepancies you implied. Passing custom `margin` and `font-size` could address that, but I wonder if we are attempting to prematurely force an abstraction (i.e. is there a single badge style that we'd like to apply everywhere or would adding diverging styling options make the abstraction unnecessarily complex)? Additionally, I noticed that the `Badge` component doesn't have any dark-mode specific styles, and this block type list context may need those. So, it may require some coordination to see if the badge styles do align well between this block types list and the featured image. 

All that said, I'll definitely take another look to see if sharing the code makes sense. 

@kyleaparker WDYT about aligning the styles between the featured image and block type list badges? It would appear the padding, font size, background, and border radius may be a little bit different, as the block type list is more space constrained and requires a smaller badge. Do we have a desire to standardize this badge or should we treat the block type list as a separate pattern? Additionally, do we need dark-mode styles for the badge? 

| Featured Image | Block Type List |
| --- | --- |
| <img width="388" alt="featured-image-badge" src="https://user-images.githubusercontent.com/438664/124153093-b2b3bb80-da59-11eb-8bd2-d5619c5007d2.png"> | <img width="599" alt="block-type-list-baddge" src="https://user-images.githubusercontent.com/438664/124153132-b8a99c80-da59-11eb-8cf8-748594a1e83c.png"> |

Reply:
Thanks, David! Looking forward to following the discussion here and would be happy to help in any way if there's a decision to align the badge styles. I also wanted to note that the padding in the "featured" badge has been reduced since that screenshot, following design feedback in https://github.com/WordPress/gutenberg/pull/32865, here's an updated screenshot for comparison:

![122811271-2c181500-d2c8-11eb-9f3f-1b9604877b29](https://user-images.githubusercontent.com/2998162/124255583-3c2ec080-db22-11eb-9759-f5245c0d44e9.png)

Reply:
@dcalhoun let's align on the styles - I think it's better they match :)

Reply:
> I also wanted to note that the padding in the "featured" badge has been reduced since that purple screenshot, following design feedback in #32865 [...]

@SiobhyB ah, thank you for sharing that. I was unaware of that. It likely changed since the first time I explored reusing the `Badge` component. 

> @dcalhoun let's align on the styles - I think it's better they match :)

@kyleaparker sounds good. I suppose I'll use my discretion in modifying the styles to work in both contexts and dark mode; and let you know if I run into any specific issues.